configs/ruby/Ruby.py

```diff
@@ -307,9 +307,13 @@ def create_system(
 def create_directories(options, bootmem, ruby_system, system):
     import importlib
 
+    if "PROTOCOL" not in buildEnv:
+        raise RuntimeError(
+            "The 'PROTOCOL' key is missing in buildEnv. Please ensure it is set correctly."
+        )
     Directory_Controller = getattr(
         importlib.import_module("m5.objects"),
-        f"{options.protocol}_Directory_Controller",
+        f"{buildEnv['PROTOCOL']}_Directory_Controller",
```

Comment:
Isn't this against the multi-ruby support? What happens to the PROTOCOL env var if gem5 is built with all protocols?

IMHO it's better to  check for options.protocol and only if this is not set rely on buildEnv...

Reply:
I've updated this with Jason's help. The new code now supports both the new way and the "old way" as a fallback when running a single-protocol build. So with `ALL` you must use `--protocol` but it's not necessary for cases (like in the case of the failing tests) where the binary contained a single protocol.